sympy/functions/elementary/integers.py

```diff
@@ -162,13 +165,16 @@ def _eval_as_leading_term(self, x, logx=None, cdir=0):
     def _eval_nseries(self, x, n, logx, cdir=0):
         arg = self.args[0]
         arg0 = arg.subs(x, 0)
+        r = self.subs(x, 0)
+        if arg0 is S.NaN:
+            arg0 = arg._eval_nseries(x, n, logx, cdir).subs(x, 0)
```

Comment:
What is the purpose of starting with a series expansion? `subs(x, 0)` will wipe out all positive powers of `x` and only the constant term will remain (unless there are negative powers).

Reply:
I mean why not , I somewhat like the idea of having a series here . This is because of two things 

1) Firstly we address this , I don't think a series through floor should give out `nan`
```
>>> floor(sin(x)/x).series(x, 0, 100, cdir = 1)  
nan
```

2) Secondly as leading term didn't have this case already implemented all wrong results were through gruntz , so having a series would help if anytime the leading term implementation would be caught up due to something
```
>>>gruntz(floor(sin(x)/x), x, 0)
nan                  # on master
>>> gruntz(floor(sin(x)/x), x, 0)
0
```
And the initial idea behind using `subs(x, 0)` was through a reply of yours 

The question I had asked was 
```
>>> # does it make sense to return series expansion of expr inside floor ??
>>> floor(sin(x)/x).series(x)
floor(-x**2/6 + x**4/120 + O(x**6)) + 1
```
To which you had replied 
> I think that the series expansions of floor and ceiling should contain only one constant term, no powers of x.

Reply:
If I remember correctly, even wolfram does plot 0 to be the series approximation or expansion for something like `floor(sin(x)/x)` when expanded till order 20 . So I'm not fully sure how we could change what I've implemented and possibly return an alternative to this . **Do you see any other approach which we could adapt here ??**

I'll possibly have a look regarding what other CAS would return in this case !

Reply:
I would consider using `limit'.

Reply:
Ohhh ok do you mean something like this rather than using `_eval_nseries` and then doing the `subs(x, 0)` step. 

```
        if arg0 is S.NaN:
            arg0 = arg.limit(x, 0, dir='-' if re(cdir).is_negative else '+')
            r = ceiling(arg0)
```
I guess this is what you mean , I'll make the change now !

Reply:
I've made the change, maybe this is ready to go if all checks pass here , let me know if you think we can improve on the changes made here or have some doubts regarding the current implementation .

Reply:
CC @jksuom